### ISSUE TITLE
Engine can process multi-tenant DeploymentRecord commands

### ISSUE THREAD

**Issue by koevskinikola** [camunda/zeebe#13320]
### Description

The engine should support working with multi-tenancy during deployments.

### AC
- [x] The `RecordValueWithTenant` interface (see #13235) is implemented by the following classes:
       - [ ] `DeploymentRecord`
       - [ ] `ProcessMetadata`
       - [ ] `ProcessRecord`
       - [ ] `DecisionRequirementsMetadataRecord`
       - [ ] `DecisionRequirementsRecord`
       - [ ] `DecisionRecord`
- [x] `DeploymentRecord` implements the `RecordValueWithTenantPermissions` interface (see #13235)
- [ ] ~~`DeploymentCreateProcessor` rejects the command if the `tenantId` isn't found in the list of `tenantIds` available to the user.~~
- [x] The `DeploymentCreateProcessor`/`BpmnResourceTransformer`/`DmnResourceTransformer` propagates the `tenantId` to the BPMN/DMN resources and their appropriate Records.
- [x] The `DeploymentState`, `ProcessState`, and `DecisionState`  include a `tenantId` key when an appropriate resource is added.
- [x] The unit test coverage is extended to cover the new tenant properties.

### Blocked by
- #13235

**Comment by koevskinikola** [comment]
:question: Do we need to tackle the `DeploymentDistributionRecord` as well? 

The record is deprecated, so it would only be triggered by existing "in-progress" deployments. We can classify this as old data, and map this deployment to the `<default>` tenant.

**Comment by remcowesterhoud** [comment]
> DeploymentCreateProcessor rejects the command if the tenantId isn't found in the list of tenantIds available to the user.

This will happen in the gateway, so I won't do this as part of this issue.

**Comment by remcowesterhoud** [comment]
> ❓ Do we need to tackle the DeploymentDistributionRecord as well?

I don't think this record needs it. This record doesn't know any details about a process. It was used to store a key + partition id in the state. This kept track of what deployment needed to be distributed to what partition. But the actual command that got distributed was the `Deployment.DISTRIBUTE`, which is actually the `DeploymentRecord`.

**Comment by koevskinikola** [comment]
Closing this issue as all the sub-tasks have been completed.
